MWI: Add Ansible AWX guide

timothyb89 opened the pull request:
This adds a guide for accessing nodes via a tbot sidecar in Ansible AWX.

Closes #52040

timothyb89 committed:
MWI: Add Ansible AWX access guide

This adds a guide for accessing nodes via tbot in Ansible AWX.

timothyb89 committed:
Add tbot config

timothyb89 committed:
Add missing sections, resolve TODOs

timothyb89 commented:
> I think a helpful addition would be a quick explainer of _why_ the fdpass binary is what we're recommending, and a link to the docs for it specifically. Just something like "For Ansible use cases use the [fdpass binary](https://goteleport.com/docs/reference/cli/fdpass-teleport/) because it enables large-scale multiplexing that the standard tbot binary does not."

Good call, I've tried to add a short explainer and linked to the reference docs.

timothyb89 committed:
Add cspell entries

timothyb89 committed:
Add missing steps to actually run the awx job

timothyb89 committed:
Fix various lints

boxofrad commented on docs/pages/machine-workload-identity/machine-id/access-guides/ansible-awx.mdx:
```diff
@@ -0,0 +1,446 @@
+      # join method does not require persistence.
+      type: memory
+    # ensure this is configured to the address of your Teleport Proxy Service.
+    proxy_server: <Var name="example.teleport.sh" />:443
```
nit: Should the port be inside the variable?

strideynet replied:
I seem to recall this possibly breaks the variables - but maybe we fixed that.

strideynet reviewed (commented):
Content largely looks good - before merging however, can we just do a pass of adding a little more detail throughout - especially when we provide config files/resource specs, it'd be nice to tell the user what the purpose of that resource/config will be.

zmb3 reviewed (commented):
I'm not familiar with Ansible, so maybe this is pretty standard, but my initial instinct is that this guide feels pretty complicated.

timothyb89 committed:
Address review feedback

- Adds an explicit role
- Adds descriptions for created Teleport and k8s resources
- Adds missing filenames, save steps, and other implied steps
- Fixes various typos
- Adds links to the AWX guide from the Ansible guide and list of guides

timothyb89 commented on docs/pages/machine-workload-identity/machine-id/access-guides/ansible-awx.mdx:
````diff
@@ -0,0 +1,446 @@
+```
+
+Be sure to adjust this configuration as needed:
+- Grant any desired roles to the bot by appending them to the `spec.roles` list
````
I've added an explicit role (the `*: *` example from the original ansible guide) along with a note that you can add other existing roles as well.

timothyb89 commented on docs/pages/machine-workload-identity/machine-id/access-guides/ansible-awx.mdx:
I've added links from our list of guides and from the traditional Ansible guide.

timothyb89 commented on docs/pages/machine-workload-identity/machine-id/access-guides/ansible-awx.mdx:
```diff
@@ -0,0 +1,446 @@
+spec:
+  # roles is a list of roles to grant to the Bot. These roles should grant the
+  # bot SSH access to all the nodes you want to access via your Ansible jobs.
+  roles: []
```
Per @strideynet's suggestion, I've added an explicit role along with a note that users can add additional roles as desired.

timothyb89 committed:
Fix typo in docs link

timothyb89 committed:
Add more detail to the configmap description

timothyb89 committed:
Tweak JWKS description

timothyb89 committed:
Include additional references to Ansible Automation Platform

timothyb89 commented on docs/pages/machine-workload-identity/machine-id/access-guides/ansible-awx.mdx:
```diff
@@ -0,0 +1,505 @@
+ - infrastructure-identity
+---
+
+Ansible AWX, formerly known as Ansible Tower, is an interface on top of Ansible
```
I've added a sentence below that references it properly ("Ansible Automation Platform automation controller" is... quite the product name...) and tries to explain the difference. I'm a bit wary of further overloading the first sentence but can give it another pass if you think it's important to include AAP(ac) from the start.

strideynet replied:
I think as long as it's within the text of the document a few times - I'm happy - will just improve discoverability.

timothyb89 commented on docs/pages/machine-workload-identity/machine-id/access-guides/ansible-awx.mdx:
```diff
@@ -0,0 +1,446 @@
+we'll assume your AWX jobs will run on a Kubernetes cluster where bots can join
+using the `static_jwks` joining mode.
+
+<Admonition type="tip" title="OIDC Joining">
```
Good call, issue here: https://github.com/gravitational/teleport/issues/59619
